src/lib/pubkey/classic_mceliece/cmce_matrix.cpp

```diff
@@ -0,0 +1,258 @@
+/*
+ * Classic McEliece Matrix Logic
+ * Based on the public domain reference implementation by the designers
+ * (https://classic.mceliece.org/impl.html - released in Oct 2022 for NISTPQC-R4)
+ *
+ *
+ * (C) 2023 Jack Lloyd
+ *     2023,2024 Fabian Albert, Amos Treiber - Rohde & Schwarz Cybersecurity
+ *
+ * Botan is released under the Simplified BSD License (see license.txt)
+ **/
+
+#include <botan/internal/cmce_matrix.h>
+
+namespace {
+using word_t = uint64_t;
+
+}  // Anonymous namespace
+
+namespace Botan {
+
+//TODO Renéquest: This file could use MatrixRow and Matrix strong types that implement .at() etc.
```

Comment:
Since we rarely access individual bits but entire rows most of the time, we decided against a bit matrix class and used a vector of bitvectors instead (one for each row).